scripts/base/frameworks/notice/actions/email_admin.zeek

```diff
@@ -18,18 +18,28 @@ export {
 	};
 }
 
-hook notice(n: Notice::Info) &priority=-5
+hook notice(n: Notice::Info)
 	{
 	if ( |Site::local_admins| > 0 &&
 	     ACTION_EMAIL_ADMIN in n$actions )
 		{
 		local email = "";
 		if ( n?$src && |Site::get_emails(n$src)| > 0 )
-			email = fmt("%s, %s", email, Site::get_emails(n$src));
+			email = Site::get_emails(n$src);
 		if ( n?$dst && |Site::get_emails(n$dst)| > 0 )
-			email = fmt("%s, %s", email, Site::get_emails(n$dst));
-		
+			{
+			if ( email != "" )
+				email = fmt("%s, %s", email, Site::get_emails(n$dst));
+			else
+				email = Site::get_emails(n$dst);
+			}
+
 		if ( email != "" )
-			email_notice_to(n, email, T);
+			{
+			if ( ! n?$email_dest )
+				n$email_dest = set();
+
+			add n$email_dest[email];
+			}
```

Comment:
Could the two `if` blocks that currently set `email` add to `email_dest` directly, instead of building the string?

Reply:
I'm trying to duplicate the current functionality of `EMAIL_ADMIN`, and I can take a guess as to why it's like that. `Site::local_admins` allows for multiple e-mails for a given subnet. I think in many cases, it'd be useful for those admins to be on a single e-mail. It cuts down on the number of sendmail invocations, any one admin can see which of their peers was also notified, and they can reply-all to ensure that all the admins see the discussion.

At the same time, if we have admins for both src and dst, I think it makes sense to receive separate e-mails. Admin Alice maybe doesn't need to know that Admin Bob was the one to blame.

Reply:
Actually, I just realized that the way I've written it, Alice and Bob would be on the same e-mail. So, yes, I think your suggestion is the right approach. Let me write a test for this and see what I can simplify.

Reply:
Does it need to be special-cased? I could also see supporting the more generic use-case:

``` zeek
local a: set[string];

add a["foo"];
```